### ISSUE TITLE
Autoregressive evaluation computes probability loss on decoded character IDs

### ISSUE THREAD

**Issue by nm-z** [nm-z/nates-recipe-rs#104]
## Problem

Autoregressive evaluation reports a large negative `ce` or `bce` loss because
it applies the probability-loss function to predictions and targets after both
have been decoded onto the raw character-ID scale.

Verified on `origin/minimal` at `52863491` through the public `.split(...)`
evaluation path.

## Reproduction

```rust
use recipe::*;

fn main() {
	for split in [1.0, 0.8] {
		let data = recipe.data(auto).set("data/auto.txt").split(split);
		let model = recipe.model().layer(16).relu().layer(16).loss(ce);
		let report = recipe.train().seed(9).epochs(150).stop(0.0).run(&model, &data);
		println!("{split} {} {}", report.initial_loss(), report.final_loss());
	}
}
```

The full-training path reported final loss `0.145297`. Adding `.split(0.8)`
reported `ce -2690.2005` while initial loss remained `0.640923`. The same branch
produced negative BCE loss.

## Cause

The autoregressive evaluation branch calls `model_loss` after
`TargetScale::decode` has converted predictions to raw character IDs, and it
passes raw character-ID targets (`recipe.rs:8459-8468`). The probability loss
expects a raw logit and a target encoded into `[0,1]`.

## Required behavior

- Evaluate loss in the same representation used by training.
- Keep initial, training, and evaluation losses comparable and non-negative for
  probability losses.
- Preserve decoding only for user-facing predictions and tokens.
